airflow/www_rbac/views.py

```diff
@@ -1856,47 +1919,118 @@ class DagRunModelView(AirflowModelView):
+    @provide_session
+    def action_set_failed(self, drs, session=None):
+        try:
+            DR = models.DagRun
+            count = 0
+            dirty_ids = []
+            altered_tis = []
+            for dr in session.query(DR).filter(
+                DR.id.in_([dagrun.id for dagrun in drs])).all():
+                dirty_ids.append(dr.dag_id)
+                count += 1
+                altered_tis += \
+                    set_dag_run_state_to_failed(dagbag.get_dag(dr.dag_id),
+                                                dr.execution_date,
+                                                commit=True,
+                                                session=session)
+            models.DagStat.update(dirty_ids, session=session)
+            altered_ti_count = len(altered_tis)
+            flash(
+                "{count} dag runs and {altered_ti_count} task instances "
+                "were set to failed".format(**locals()))
+        except Exception as ex:
+            flash('Failed to set state', 'error')
+        return redirect(self.route_base + '/list')
 
+    @action('set_success', "Set state to 'success'",
+            "All task instances would also be marked as success, are you sure?",
+            single=False)
     @provide_session
-    def set_dagrun_state(self, drs, target_state, session=None):
+    def action_set_success(self, drs, session=None):
         try:
             DR = models.DagRun
             count = 0
             dirty_ids = []
+            altered_tis = []
             for dr in session.query(DR).filter(
-                    DR.id.in_([dagrun.id for dagrun in drs])).all():
+                DR.id.in_([dagrun.id for dagrun in drs])).all():
                 dirty_ids.append(dr.dag_id)
                 count += 1
-                dr.state = target_state
-                if target_state == State.RUNNING:
-                    dr.start_date = timezone.utcnow()
-                else:
-                    dr.end_date = timezone.utcnow()
-            session.commit()
+                altered_tis += \
+                    set_dag_run_state_to_success(dagbag.get_dag(dr.dag_id),
+                                                 dr.execution_date,
+                                                 commit=True,
+                                                 session=session)
             models.DagStat.update(dirty_ids, session=session)
+            altered_ti_count = len(altered_tis)
             flash(
-                "{count} dag runs were set to '{target_state}'".format(**locals()))
+                "{count} dag runs and {altered_ti_count} task instances "
+                "were set to success".format(**locals()))
         except Exception as ex:
             flash('Failed to set state', 'error')
         return redirect(self.route_base + '/list')
 
+    # Called after editing DagRun model in the UI.
+    @provide_session
+    def after_model_change(self, form, dagrun, is_created, session=None):
```

Comment:
@jgao54 Hi Joy, is DAG run model edit disabled in RBAC view even for Admin? If so I'll remove this part of change

Reply:
One additional note is that `after_model_change` is a Flask-Admin feature and will not work with RBAC.

Reply:
@jgao54 excellent, I'm against the edit power too. Tyvm for the RBAC work again then, we'd love to migrate to it soon.